src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -211,4 +234,47 @@ private void validateMedicalInventory(MedicalInventory medInventory) throws OHDa
 			throw new OHDataValidationException(errors);
 		}
 	}
+	
+	@Transactional(rollbackFor = OHServiceException.class)
+	public boolean validateInventory(MedicalInventory inventory, List<MedicalInventoryRow> inventoryRowSearchList) throws OHServiceException {
+		String dischargeCode = inventory.getDischargeType();
+		String chargeCode = inventory.getChargeType();
+		Integer supplierId = inventory.getSupplier();
+		String wardCode = inventory.getDestination();
+		MovementType chargeType = medicalDsrStockMovementTypeBrowserManager.getMovementType(chargeCode);
+		MovementType dischargeType = medicalDsrStockMovementTypeBrowserManager.getMovementType(dischargeCode);
+		Supplier supplier = supplierManager.getByID(supplierId);
+		Ward ward = wardManager.findWard(wardCode);
+		LocalDateTime today = LocalDateTime.now();
+		for (Iterator<MedicalInventoryRow> iterator = inventoryRowSearchList.iterator(); iterator.hasNext();) {
+			String reference = "Ref-"+LocalDateTime.now();
```

Comment:
Is there any possibility of generating the same `reference` string if the code runs very, very fast?